### PR TITLE
fix: adds support for builds with cache

### LINKED ISSUE #3: Plugin is not working with webpack cache
Hi! Thx for the plugin, it helps a lot to use pino with typescript in production code.
But there is a problem with a cache. If you will add:

```js
cache: {
        name: 'test',
        type: 'filesystem',
        buildDependencies: {
          config: [__filename]
        }
      },
```
to your webpack config, the first test will be ok, but the second will fail. The problem is, that plugin won't be executed, if webpack uses a code from cache. And I do not know, how to fix it(

Of course, I can use in memory cache, but the first build will take much time.

By the way, in memory cache will break the code too. If I change a transport — I will see an error:
```
logServerUncaughtException error:  Error: The worker thread exited
    at Worker.onWorkerExit (/Users/a.malko/Work/react-ssr-template/build/main.js:95787:32)
    at Worker.emit (node:events:390:28)
    at Worker.[kOnExit] (node:internal/worker:278:10)
    at Worker.<computed>.onexit (node:internal/worker:198:20)
/Users/a.malko/Work/react-ssr-template/build/main.js:95821
      throw new Error('the worker has exited')
      ^

Error: the worker has exited
    at ThreadStream.write (/Users/a.malko/Work/react-ssr-template/build/main.js:95821:13)
    at Pino.write (/Users/a.malko/Work/react-ssr-template/build/main.js:99370:10)
    at Pino.LOG [as error] (/Users/a.malko/Work/react-ssr-template/build/main.js:99677:21)
    at logServerUncaughtException (/Users/a.malko/Work/react-ssr-template/build/main.js:10882:66)
    at process.<anonymous> (/Users/a.malko/Work/react-ssr-template/build/main.js:101485:133)
    at process.emit (node:events:390:28)
    at process._fatalException (node:internal/process/execution:170:25)
```

### PR DESCRIPTION
**Pull request by gdorsi**
fixes #3

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: test/index.test.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ test('it should correctly generated all required pino files', (t) => {
   const distFolder = resolve(__dirname, '../tmp/dist')
 
   t.teardown(() => {
-    spawnSync(`rm -rf ${distFolder}`)
+    spawnSync(`rm -rf ${distFolder}`, {
+      shell: true
```
Comment (simoneb) [inline test/index.test.js:17]: why this change?

File: test/cached-build.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      plugins: [new PinoWebpackPlugin({ transports: ['pino-pretty'] })],
+      optimization: {
+        minimize: false
+      }
+    },
+    onDone
+  )
+}
+
+test('it should correctly generated all required pino files', (t) => {
```
Comment (simoneb) [inline test/cached-build.test.js:41]:
```suggestion
test('it should correctly generate all required pino files', (t) => {
```

File: test/cached-build.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  const distFolder = resolve(__dirname, '../tmp/cached-build')
+
+  t.teardown(() => {
+    spawnSync(`rm -rf ${distFolder}`, {
+      shell: true
+    })
+  })
+
+  runBuild(distFolder, () => {
```
Comment (simoneb) [inline test/cached-build.test.js:52]: add a check that this doesn't error

File: test/cached-build.test.js (hunk 2 above)
Comment (simoneb) [reply test/cached-build.test.js:41]: How is this testing that the output is being cached? It looks like it's testing that the output is generated correctly the second time you run the build, but not that the cache is being used. Am I missing something?

File: test/index.test.js (hunk 1 above)
Comment (gdorsi) [reply test/index.test.js:17]: That's because the command didn't locate correctly the `rm` executable so the tmp dir was not deleted.

File: test/cached-build.test.js (hunk 2 above)
Comment (gdorsi) [reply test/cached-build.test.js]: The caching is enabled in the webpack configuration and I'm testing that nothing breaks when it is enabled.

I've renamed the test to make the intent more clear.

File: test/cached-build.test.js (hunk 3 above)
Comment (gdorsi) [reply test/cached-build.test.js]: done

**Review by simoneb — APPROVED** [review]

**Comment by artem-malko** [comment]
@mcollina lets publish it?)
